prometheus-to-sd/main.go

```diff
@@ -136,6 +138,10 @@ func readAndPushDataToStackdriver(stackdriverService *v3.Service, gceConf *confi
 			glog.Warningf("Error while getting Prometheus metrics %v", err)
 			continue
 		}
+		if *squashComponentName {
+			metrics = translator.SquashComponentName(metrics, sourceConfig.Component)
+			sourceConfig.SquashComponentName(sourceConfig.Component)
```

Comment:
I think we can assume that the metrics are already whitelisted in shortened form so there's no need to trim the prefix there too.